x-pack/plugins/endpoint/server/config.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { schema, TypeOf } from '@kbn/config-schema';
+import { Observable } from 'rxjs';
+import { PluginInitializerContext } from 'kibana/server';
+
+export type EndpointConfigType = ReturnType<typeof createConfig$> extends Observable<infer P>
```

Comment:
Would you explain this? If I'm understanding this correctly, `createConfig$` always returns a value that extends Observable, so the second part of this expression is unused.

Reply:
I followed some other examples that are in master already, https://github.com/elastic/kibana/blob/master/x-pack/plugins/spaces/server/config.ts, https://github.com/elastic/kibana/blob/master/x-pack/plugins/security/server/config.ts